keanu-project/src/test/java/io/improbable/keanu/vertices/dbltensor/probabilistic/TensorGammaVertexTest.java

```diff
@@ -0,0 +1,232 @@
+package io.improbable.keanu.vertices.dbltensor.probabilistic;
+
+import io.improbable.keanu.distributions.continuous.Gamma;
+import io.improbable.keanu.vertices.dbl.probabilistic.GammaVertex;
+import io.improbable.keanu.vertices.dbltensor.*;
+import io.improbable.keanu.vertices.dbltensor.nonprobabilistic.ConstantTensorVertex;
+import io.improbable.keanu.vertices.dbltensor.nonprobabilistic.diff.TensorPartialDerivatives;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.Supplier;
+
+import static io.improbable.keanu.vertices.dbltensor.probabilistic.ProbabilisticDoubleTensorContract.moveAlongDistributionAndTestGradientOnARangeOfHyperParameterValues;
+import static org.junit.Assert.assertEquals;
+
+public class TensorGammaVertexTest {
+
+    private KeanuRandom random;
+
+    private static final double DELTA = 0.0001;
+
+    @Before
+    public void setup() {
+        random = new KeanuRandom(1);
+    }
+
+    @Test
+    public void matchesKnownLogDensityOfScalar() {
+        GammaVertex gamma = new GammaVertex(0.5, 1, 1.5, new Random(1));
+        TensorGammaVertex tensorGamma = new TensorGammaVertex(0.5, 1, 1.5, new KeanuRandom(1));
+
+        double expectedDensity = gamma.logPdf(0.5);
+
+        ProbabilisticDoubleTensorContract.matchesKnownLogDensityOfScalar(tensorGamma, 0.5, expectedDensity);
+    }
+
+    @Test
+    public void matchesKnownLogDensityOfVector() {
+        double expectedLogDensity = Gamma.logPdf(0.0, 1.0, 5., 1) + Gamma.logPdf(0.0, 1.0, 5., 3);
+        TensorGammaVertex tensorGamma = new TensorGammaVertex(0.0, 1., 5., new KeanuRandom(1));
+        ProbabilisticDoubleTensorContract.matchesKnownLogDensityOfVector(tensorGamma, new double[]{1., 3.}, expectedLogDensity);
+    }
+
+    @Test
+    public void matchesKnownDerivativeLogDensityOfScalar() {
+        KeanuRandom keanuRandom = new KeanuRandom(1);
+
+        Gamma.Diff gammaLogDiff = Gamma.dlnPdf(0.75, 2, 5.5, 1.5);
+
+        TensorUniformVertex aTensor = new TensorUniformVertex(new ConstantTensorVertex(0.5), new ConstantTensorVertex(1.0), keanuRandom);
```

Comment:
Do you need the ConstantTensorVertex wrappers? Or do the various constructors help you out?

Reply:
Good spot, had loads of them scattered around. Removed them